### DISCUSSION TITLE
Lazarus + Linux + Indy, does the bundle work?

### DISCUSSION THREAD

**Discussion by Molochnik** [IndySockets/Indy#603] [Q&A]
I've made a small test app and tried to test TIdTCPServer+TIdTCPClient and TIdUDPServer+TIdUDPClient connections for exchanging data. The code works fine in Delphi (Windows and Linux) and in Lazarus (Windows only), Lazarus on Linux  works with errors, TIdIOHandler.ReadStream cannot detect available data, in UDP connection some weird protocol number somehow appears (neither ipv4 nor ipv6). Does it make sense to create an issue providing the sample app or better just accept that Indy doesn't work properly with Lazarus on Linux?

**Comment by rlebeau** [comment]
Please open an issue ticket, Indy has supported Linux on FreePascal/Lazarus for a long time and all of that stuff should be working.

**Comment by rlebeau** [comment]
@Molochnik  Is this still an open issue for you?

**Comment by Molochnik** [comment]
No, everything is fine, close the issue! Thanks!
